cluster-autoscaler/core/scaleup/orchestrator/orchestrator_test.go

```diff
@@ -761,6 +763,141 @@ func TestScaleUpNoHelp(t *testing.T) {
 	assert.Regexp(t, regexp.MustCompile("NotTriggerScaleUp"), event)
 }
 
+type constNodeGroupSetProcessor struct {
+	similarNodeGroups []cloudprovider.NodeGroup
+}
+
+func (p *constNodeGroupSetProcessor) FindSimilarNodeGroups(_ *context.AutoscalingContext, _ cloudprovider.NodeGroup, _ map[string]*schedulerframework.NodeInfo) ([]cloudprovider.NodeGroup, errors.AutoscalerError) {
+	return p.similarNodeGroups, nil
+}
+
+func (p *constNodeGroupSetProcessor) BalanceScaleUpBetweenGroups(_ *context.AutoscalingContext, _ []cloudprovider.NodeGroup, _ int) ([]nodegroupset.ScaleUpInfo, errors.AutoscalerError) {
+	return nil, nil
+}
+
+func (p *constNodeGroupSetProcessor) CleanUp() {}
+
+func TestComputeSimilarNodeGroups(t *testing.T) {
+	pod1 := BuildTestPod("p1", 100, 1000)
+	pod2 := BuildTestPod("p2", 100, 1000)
+	pod3 := BuildTestPod("p3", 100, 1000)
+
+	testCases := []struct {
+		name                  string
+		nodeGroup             string
+		similarNodeGroups     []string
+		balancingEnabled      bool
+		schedulablePods       map[string][]*apiv1.Pod
+		wantSimilarNodeGroups []string
+	}{
+		{
+			name:                  "no similar node groups",
+			nodeGroup:             "ng1",
+			balancingEnabled:      true,
+			wantSimilarNodeGroups: []string{},
+		},
+		{
+			name:                  "some similar node groups, but no schedulable pods",
+			nodeGroup:             "ng1",
+			similarNodeGroups:     []string{"ng2", "ng3"},
+			balancingEnabled:      true,
+			wantSimilarNodeGroups: []string{},
+		},
+		{
+			name:              "some similar node groups and same schedulable pods, but balancing disabled",
+			nodeGroup:         "ng1",
+			similarNodeGroups: []string{"ng2", "ng3"},
+			balancingEnabled:  false,
+			schedulablePods: map[string][]*apiv1.Pod{
+				"ng1": {pod1},
+				"ng2": {pod1},
+				"ng3": {pod1},
+			},
+			wantSimilarNodeGroups: []string{},
+		},
+		{
+			name:              "some similar node groups and same schedulable pods",
+			nodeGroup:         "ng1",
+			similarNodeGroups: []string{"ng2", "ng3"},
+			balancingEnabled:  true,
+			schedulablePods: map[string][]*apiv1.Pod{
+				"ng1": {pod1},
+				"ng2": {pod1},
+				"ng3": {pod1},
+			},
+			wantSimilarNodeGroups: []string{"ng2", "ng3"},
+		},
+		{
+			name:              "similar node groups can schedule more pods",
+			nodeGroup:         "ng1",
+			similarNodeGroups: []string{"ng2", "ng3"},
+			balancingEnabled:  true,
+			schedulablePods: map[string][]*apiv1.Pod{
+				"ng1": {pod1},
+				"ng2": {pod1, pod2},
+				"ng3": {pod1, pod2, pod3},
+			},
+			wantSimilarNodeGroups: []string{"ng2", "ng3"},
+		},
+		{
+			name:              "similar node groups can schedule different/no pods",
+			nodeGroup:         "ng1",
+			similarNodeGroups: []string{"ng2", "ng3"},
+			balancingEnabled:  true,
+			schedulablePods: map[string][]*apiv1.Pod{
+				"ng1": {pod1, pod2},
+				"ng2": {pod1},
+			},
+			wantSimilarNodeGroups: []string{},
+		},
+	}
```

Comment:
Is it worth adding a case with non-similar nodeGroup that has matching schedulablePods? As is I think I could remove FindSimilarNodeGroups() call from ComputeSimilarNodeGroups() and the tests would still pass.